src/Features/CSharp/Portable/ReplaceDefaultLiteral/CSharpReplaceDefaultLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,150 @@
+            var token = syntaxRoot.FindToken(context.Span.Start);
+
+            if (token.Span == context.Span &&
+                token.IsKind(SyntaxKind.DefaultKeyword) &&
+                token.Parent.IsKind(SyntaxKind.DefaultLiteralExpression))
+            {
+                var defaultLiteral = (LiteralExpressionSyntax)token.Parent;
+                var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                var (newExpression, displayText) = GetReplacementExpressionAndText(
+                    context.Document, defaultLiteral, semanticModel, context.CancellationToken);
+
+                if (newExpression != null)
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(
+                            c => ReplaceAsync(context.Document, context.Span, newExpression, c),
+                            displayText),
+                        context.Diagnostics);
+                }
+            }
+        }
+
+        private static async Task<Document> ReplaceAsync(
+            Document document, TextSpan span, SyntaxNode newExpression, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var defaultToken = syntaxRoot.FindToken(span.Start);
+            var defaultLiteral = (LiteralExpressionSyntax)defaultToken.Parent;
+
+            var newRoot = syntaxRoot.ReplaceNode(defaultLiteral, newExpression.WithTriviaFrom(defaultLiteral));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private static (SyntaxNode newExpression, string displayText) GetReplacementExpressionAndText(
+            Document document,
+            LiteralExpressionSyntax defaultLiteral,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var type = semanticModel.GetTypeInfo(defaultLiteral, cancellationToken).ConvertedType;
+            if (type != null && type.TypeKind != TypeKind.Error)
+            {
+                if (IsFlagsEnum(type, semanticModel.Compilation) &&
+                    type.GetMembers("None").FirstOrDefault() is IFieldSymbol field && IsZero(field.ConstantValue))
+                {
+                    return GenerateMemberAccess("None");
+                }
+                else if (type.Equals(semanticModel.Compilation.GetTypeByMetadataName("System.Threading.CancellationToken")))
+                {
+                    return GenerateMemberAccess(nameof(CancellationToken.None));
+                }
+                else if (type.SpecialType == SpecialType.System_IntPtr || type.SpecialType == SpecialType.System_UIntPtr)
+                {
+                    return GenerateMemberAccess(nameof(IntPtr.Zero));
```

Comment:
fwiw, this was a bit confusing to me because it makes it seem as if UintPtr will be rewritten to IntPtr.Zero.  This is generally why i'm not a huge fan of capturing local functions.  I can live with it here, but my preference would be to pass the type to make things clear: "Oh, ok, i'm accessing 'Zero' off the type actually passed in". 

Your call! :)

Reply:
I see your point and I would fix that and make it a method if all I had to pass in was the type, but it also captures `generator`. Won't fix.